mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala

```diff
@@ -184,16 +184,16 @@ object Tool extends LazyLogging {
   private def runAndExit(
       executor: PassChainExecutor,
       msgIfOk: TlaModule => String,
-      msgIfFail: String,
-      errCode: Int = ExitCodes.ERROR): Int = {
+      msgIfFail: String): ExitCodes.TExitCode = {
     val result = executor.run()
-    if (result.isDefined) {
-      logger.info(msgIfOk(result.get))
-      ExitCodes.OK
-    } else {
-      logger.info(msgIfFail)
-      errCode
-    }
+    result.fold(errorCode => {
+          logger.info(msgIfFail)
+          errorCode
+        },
+        module => {
+          logger.info(msgIfOk(module))
+          ExitCodes.OK
+        })
```

Comment:
Have to say, I feel like a case analysis using `match` would be clearer here:

```suggestion
    result match {
      case Left(errorCode) => {
          logger.info(msgIfFail)
          errorCode
        }
      case Right(module) => {
          logger.info(msgIfOk(module))
          ExitCodes.OK
        })
```

My sense is that the `fold` cominator only aids readability when you have some other combinators to feed into it, or you are ignoring the values. Otherwise you just end up with an explicit case analysis only it's positional instead of using pattern matching to make it explicit what value on the lhs gets eliminated by what expression on the rhs.

WDYT?

Reply:
This is the one place I'd really prefer to see a change, unless you feel strongly that it isn't an improvement.

Reply:
> only it's positional instead of using pattern matching to make it explicit what value on the lhs gets eliminated

To me personally, `errorCode` and `module` make it pretty obvious which case is which, and the fold is syntactically less cluttered. I don't have a strong preference here, so I'm going include your suggestion.

Reply:
I can see what you’re saying here. the perceived readability probably just comes from familiarity on my end. I guess I still think this parses easier given the rest of our codebase, but wouldn’t object if you d coded to change it back. :)

Reply:
I think this really depends on what you're reading the code for. Clearly marking `Left` and `Right` has its value, and it makes the type of `result` obvious without a type annotation, so I'm quite happy to keep your suggestion :)